src/tests/encore/parametric/methodPolyClosure.enc

```diff
@@ -0,0 +1,18 @@
+read class Bar[t]
+  def bar() : unit
+    println("In Bar")
+  end
+end
+
+read class Foo
+  def foo[ty]() : () -> unit
+    fun() => (new Bar[ty]()).bar()
```

Comment:
@EliasC What is the status of this PR? Does it still fail for @kikofernandez's example?